### ISSUE TITLE
Magnet URI will save to session file, but biTtorrent file will not save to session file

### ISSUE THREAD

**Issue by Aaron20127** [aria2/aria2#670]
Hi
I use aria2-1.19.0 to dowaload a Magnet URI. My options are

> bt-save-metadata=true
force-save=true
save-session-interval=120
bt-metadata-only=false
input-file=/etc/plugins/aria2/aria2.session
save-session=/etc/plugins/aria2/aria2.session

When the MetaData file downloded, the torrent file which comes from  the MedaData will start to download. But save-session option will not save this torrent file to session file, it will only save Magnet URI to the session file. Thus, when I restart aria2 process, the MetaData file will download again, but the torrent file will not be able to continue to download unless the MetaData file downloaded again. Here is the contents of  my session file:

> magnet:?xt=urn:btih:YC4XKAM2ADIU4HIKIH64OHJRUJIE3AGG
  gid=1e2d76a244a52f3e
  dir=/mnt/share/udisk/Download
  parameterized-uri=true
  min-split-size=20971520
  user-agent=aria2/1.19.0

**Comment by tatsuhiro-t** [comment]
Yes, metadata is downloaded again.  Do you have a problem on restarting downloads?

**Comment by Twilight** [comment]
yeah I have exactly the same question, it still works but cost some extra time.
